### PR TITLE
CRM:  Change edit pages to fit Jetpack Emerald style

### PR DESCRIPTION
**Pull request by gogdzl**
## Proposed changes:
* This PR changes the following pages to fit Jetpack Emerald style:
    * Edit Contact
    * Edit Company
    * Edit Quote
    * Edit Quote Template
    * Edit Transactions 

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
[Change CRM edit pages to fit the Emerald style#3124](https://github.com/Automattic/zero-bs-crm/issues/3124)

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
* Go to the following pages and make sure it fits the latest Jetpack Emerald style:
 _(Note: test it while editing existing content, and also adding new content)_
    * Edit Contact
    * Edit Company
    * Edit Quote
    * Edit Quote Template
    * Edit Transactions 

## Old vs New screenshots:

image
image

image
image

image
image

image
image

image
image

### CODE REVIEW

File: projects/plugins/crm/includes/ZeroBSCRM.MetaBoxes3.Companies.php [hunk 1]
```diff
@@ -115,278 +115,116 @@ public function html( $company, $metabox ) {
...
-                $showCountryFields = zeroBSCRM_getSetting('countries');
-                $second_address_label = zeroBSCRM_getSetting( 'secondaddresslabel' );
+					global $zbsCompanyFields; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
+					$fields               = $zbsCompanyFields; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
+					$show_id              = (int) $zbs->settings->get( 'showid' );
+					$fields_to_hide       = $zbs->settings->get( 'fieldhides' );
+					$show_addresses       = (int) zeroBSCRM_getSetting( 'showaddress' );
+					$show_second_address  = (int) zeroBSCRM_getSetting( 'secondaddress' );
+					$show_country_fields  = zeroBSCRM_getSetting( 'countries' );
+					$second_address_label = zeroBSCRM_getSetting( 'secondaddresslabel' );
```
Comment (tbradsha) [inline projects/plugins/crm/includes/ZeroBSCRM.MetaBoxes3.Companies.php]: Nitpick, but best to be consistent and use `$zbs->settings->get()` if possible.

File: projects/plugins/crm/includes/ZeroBSCRM.MetaBoxes3.Companies.php [hunk 2]
```diff
@@ -115,278 +115,116 @@ public function html( $company, $metabox ) {
...
 
-                    if ($zbsShowID == "1" && $companyID > 0) { ?>
-                    <tr class="wh-large"><th><label><?php echo esc_html( $this->coOrgLabel ) . ' '; esc_html_e("ID","zero-bs-crm");?>:</label></th>
-							<td style="font-size: 20px;color: black;vertical-align: top;">
-                        #<?php echo esc_html( $companyID ); ?>
-                    </td></tr>
+				if ( $show_id === 1 && $companyID > 0 ) { // phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
+					?>
+							<div class="jpcrm-form-group">
+								<label class="jpcrm-form-label" for="maintel"><?php echo esc_html( $this->coOrgLabel ) . ' '; esc_html_e( 'ID', 'zero-bs-crm' ); // phpcs:ignore Generic.Formatting.DisallowMultipleStatements.SameLine, WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase, Squiz.PHP.EmbeddedPhp.MultipleStatements ?>:</label>
```
Comment (tbradsha) [inline projects/plugins/crm/includes/ZeroBSCRM.MetaBoxes3.Companies.php]: Remove incorrect `for`:
```suggestion
								<label class="jpcrm-form-label"><?php echo esc_html( $this->coOrgLabel ) . ' '; esc_html_e( 'ID', 'zero-bs-crm' ); // phpcs:ignore Generic.Formatting.DisallowMultipleStatements.SameLine, WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase, Squiz.PHP.EmbeddedPhp.MultipleStatements ?>:</label>
```

File: projects/plugins/crm/includes/ZeroBSCRM.MetaBoxes3.Companies.php [hunk 3]
```diff
@@ -115,278 +115,116 @@ public function html( $company, $metabox ) {
...
-						esc_html( $status ) .
-						'</option>';
+					$field_group = '';
+
+					/*
+					 * We need to know if we are in the first column or not because
+					 * when we have a group (e.g. addresses), they should start in
+					 * the first column. We are using only two columns.
+					 */
+					$is_first_collumn = false;
```
Comment (tbradsha) [inline projects/plugins/crm/includes/ZeroBSCRM.MetaBoxes3.Companies.php]: Nitpick: `collumn` → `column`

File: projects/plugins/crm/includes/ZeroBSCRM.MetaBoxes3.Contacts.php [hunk 4]
```diff
@@ -166,327 +166,159 @@ public function __construct( $plugin_file ) {
...
-                    jQuery('#submitdiv').show();
+			<?php
+		endif;
 
-                    // turn off auto-complete on customer records via form attr... should be global for all ZBS record pages
-                    jQuery('#post').attr('autocomplete','off');
+		if ( $show_id === 1 && isset( $contact['id'] ) && ! empty( $contact['id'] ) ) :
+			?>
+							<div class="jpcrm-form-group">
+								<label class="jpcrm-form-label" for="maintel"><?php esc_html_e( 'Contact ID', 'zero-bs-crm' ); ?>:</label>
```
Comment (tbradsha) [inline projects/plugins/crm/includes/ZeroBSCRM.MetaBoxes3.Contacts.php]:
```suggestion
								<label class="jpcrm-form-label"><?php esc_html_e( 'Contact ID', 'zero-bs-crm' ); ?>:</label>
```

File: projects/plugins/crm/includes/ZeroBSCRM.MetaBoxes3.Quotes.php [hunk 5]
```diff
@@ -172,42 +172,33 @@ public function html( $quote, $metabox ) {
...
-
-                            ?><input type="hidden" name="zbsquoteid" value="<?php echo esc_attr( $quoteID ); ?>" /></div>
-                        </td></tr><?php
+					?>
+								<div class="jpcrm-form-group jpcrm-form-group-span-2">
+									<label class="jpcrm-form-label"><?php esc_html_e( 'Quote (ID)', 'zero-bs-crm' ); ?>:</label>
+								<?php
+							if ( empty( $quoteID ) ) { //phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
+								$quoteID = zeroBSCRM_getNextQuoteID(); //phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
+							}
```
Comment (tbradsha) [inline projects/plugins/crm/includes/ZeroBSCRM.MetaBoxes3.Quotes.php]: Legacy code note: correct me if I'm wrong, but I think this condition never is possible given the parent condition?

File: projects/plugins/crm/includes/ZeroBSCRM.MetaBoxes3.Quotes.php [hunk 6]
```diff
@@ -383,14 +378,13 @@ function zeroBSCRMJS_showContactLinkIf(contactID){
...
-                            </th>
-
-                        </tr><?php
+									</div>
+								<?php
 
                     } ?>
 
-            </table><?php 
+				</div></div>
```
Comment (tbradsha) [inline projects/plugins/crm/includes/ZeroBSCRM.MetaBoxes3.Quotes.php:376]: 🫣

File: projects/plugins/crm/includes/ZeroBSCRM.MetaBoxes3.Transactions.php [hunk 7]
```diff
@@ -162,24 +162,28 @@ public function html( $transaction, $metabox ) {
...
-                        }
-                    }  
+						.zbs-infobox {
+							margin: 0px !important;
+							padding: 0px !important;
+						}
+
+						.zbs-infobox.zbs-live i.zbs-help-ico {
+							font-size: unset;
+						}
```
Comment (tbradsha) [inline projects/plugins/crm/includes/ZeroBSCRM.MetaBoxes3.Transactions.php]: Maybe add a bit of specificity and move to `sass/ZeroBSCRM.admin.transactionedit.scss`?

File: projects/plugins/crm/includes/ZeroBSCRM.MetaBoxes3.Quotes.php [hunk 8]
```diff
@@ -218,16 +209,22 @@ public function html( $quote, $metabox ) {
...
-                    #} ALSO customer assigning is seperate:
-                    ?><tr class="wh-large"><th><label><?php esc_html_e('Contact',"zero-bs-crm");?>:</label></th>
-                    <td><?php
+				?>
+						<style>
+							.zbstypeaheadwrap {
+								padding-top: 0px;
+								padding-bottom: 0px;
+							}
+						</style>
```
Comment (tbradsha) [inline projects/plugins/crm/includes/ZeroBSCRM.MetaBoxes3.Quotes.php]: Maybe add a bit of specificity and move to `sass/ZeroBSCRM.admin.quotebuilder.scss`?

File: projects/plugins/crm/js/ZeroBSCRM.admin.transactioneditor.js [hunk 9]
```diff
@@ -354,7 +354,7 @@ function zeroBSCRMJS_showContactLinkIf( contactID ) {
 
 			// ALSO show in header bar, if so
 			var navButton =
-				'<a target="_blank" style="margin-left:6px;" class="zbs-trans-quicknav-contact ui icon button blue mini labeled" href="' +
+				'<a target="_blank" style="margin-left:6px;" class="zbs-trans-quicknav-contact ui icon button black mini labeled" href="' +
```
Comment (tbradsha) [inline projects/plugins/crm/js/ZeroBSCRM.admin.transactioneditor.js:357]: Nice, this was on my list.

**Review by tbradsha — COMMENTED** [review]
A few comments, but overall looks like it's what we intended! Additional notes:

* Feel free to use `jpcrm_disable_browser_autocomplete()` instead of the `<?php echo esc_attr( time() ); ?>-<?php echo esc_attr( rand(0,100) ); ?>` that is frequently used.
* I was worried about the changes in `zeroBSCRM_html_editField()` affecting other extensions, but it does appear that function is only used in core.

File: projects/plugins/crm/includes/ZeroBSCRM.MetaBoxes3.Quotes.php (hunk 5 above)
Comment (gogdzl) [reply projects/plugins/crm/includes/ZeroBSCRM.MetaBoxes3.Quotes.php]: You are right.

File: projects/plugins/crm/includes/ZeroBSCRM.MetaBoxes3.Companies.php (hunk 1 above)
Comment (gogdzl) [reply projects/plugins/crm/includes/ZeroBSCRM.MetaBoxes3.Companies.php]: Ok! =)

File: projects/plugins/crm/includes/ZeroBSCRM.MetaBoxes3.Transactions.php (hunk 7 above)
Comment (gogdzl) [reply projects/plugins/crm/includes/ZeroBSCRM.MetaBoxes3.Transactions.php]: Done.

**Review by tbradsha — APPROVED** [review]
Make it so!

**Pull request merged**